### PR TITLE
Update app-neovim.sh - build from binaries

### PR DESCRIPTION
**Pull request by Salanoid**
Build neovim from binaries because in apt last stable version is missing sometimes. 

Build made accordingly to this readme:
https://github.com/neovim/neovim/blob/master/BUILD.md

### CODE REVIEW

**Comment by Salanoid** [comment]
> @Salanoid for me eza and alacritty (among others) are also missing from apt (no install candidate). Do you have any insight as to way that might be?

This thread might be helpful: [https://github.com/alacritty/alacritty/issues/3863](https://github.com/alacritty/alacritty/issues/3863).

You can try running `apt-cache search alacritty` to look for the packages.

From what I've seen, you need to add an unofficial existing PPA `sudo add-apt-repository ppa:aslatter/ppa` to install Alacritty on different Ubuntu versions because it isn't available in the standard repositories.

File: install/app-neovim.sh [hunk 1]
```diff
@@ -1,6 +1,11 @@
-sudo add-apt-repository -y ppa:neovim-ppa/stable
-sudo apt update
-sudo apt install -y neovim
+git clone https://github.com/neovim/neovim
```
Comment (rajraj) [inline install/app-neovim.sh:1]: May be `cd` into the Downloads folder first?

Comment (Salanoid) [reply install/app-neovim.sh:2]: Yeah, we can de definitely do that. I updated my code, we also remove the repository after the installation and get back to `~/`.

**Comment by dhh** [comment]
Don't love the idea of having to build from source. Maybe @tjdevries has an opinion here?

**Comment by iljo-dp** [comment]
I'd rather instead of building from source build from neovim's prebuild binaries like this(in nvim's INSTALL.MD)

```bash
curl -LO https://github.com/neovim/neovim/releases/latest/download/nvim-linux64.tar.gz
sudo rm -rf /opt/nvim
sudo tar -C /opt -xzf nvim-linux64.tar.gz
```
And then in the shell config make sure we add that to the path(bash example)
```bash
export PATH="$PATH:/opt/nvim-linux64/bin"
```
This way we don't need to build from source but instead form nvim's own prebuild's, which is the way they recommend us installing it(and how I install it myself)

**Comment by iljo-dp** [comment]
I propose this

```bash
#!/bin/bash

# Neovim installation
cd ~/Downloads
wget -O nvim-linux64.tar.gz "https://github.com/neovim/neovim/releases/latest/download/nvim-linux64.tar.gz"
sudo rm -rf /opt/nvim
sudo tar -C /opt -xzf nvim-linux64.tar.gz
export PATH="$PATH:/opt/nvim-linux64/bin"
cd -

if [ ! -d "$HOME/.config/nvim" ]; then
	git clone https://github.com/LazyVim/starter ~/.config/nvim
	cp ~/.local/share/omakub/themes/neovim/tokyo-night.lua ~/.config/nvim/lua/plugins/theme.lua
fi
```
And also change the bashrc to include
```bash
export PATH="$PATH:/opt/nvim-linux64/bin"
```

**Comment by dhh** [comment]
Too much. The path in the app-zellij.sh installer is the way, if we go with these prebuilt binaries.

Which does bring up a bigger point that we need to eventually get this project onto being ready for arm64 too. Snapdragon is coming!

**Comment by iljo-dp** [comment]
> Too much. The path in the app-zellij.sh installer is the way, if we go with these prebuilt binaries.
> 
fair enough how's this(tested and verified this works super super fast)

```bash
cd ~/Downloads
wget -O nvim-linux64.tar.gz "https://github.com/neovim/neovim/releases/latest/download/nvim-linux64.tar.gz"
sudo rm -rf /usr/local/bin/nvim  # Remove any existing binary to avoid conflicts
sudo tar -C /usr/local/bin -xzf nvim-linux64.tar.gz --strip-components=2 nvim-linux64/bin/nvim
cd -

if [ ! -d "$HOME/.config/nvim" ]; then
	git clone https://github.com/LazyVim/starter ~/.config/nvim
	cp ~/.local/share/omakub/themes/neovim/tokyo-night.lua ~/.config/nvim/lua/plugins/theme.lua
fi
```

**Comment by dhh** [comment]
Let's follow the exact same procedure, using that install command, etc. The less variation we have between the installers, the better.

**Comment by Salanoid** [comment]
> Let's follow the exact same procedure, using that install command, etc. The less variation we have between the installers, the better.

Following the discussion, I made a new commit; we will install Neovim using the same procedure as for Zellij.

**Comment by iljo-dp** [comment]
looks better then my solution, and works great, thanks for the commit

**Comment by Salanoid** [comment]
Thank you and thanks for the help!

**Comment by dhh** [comment]
Seems like nvim 0.10 has incompatibilities with the lazyvim setup we're using. When upgrading, I get this:

```
│Error detected while processing /home/dhh/.config/nvim/init.lua:                                                                                                                                     │
│E5113: Error while calling lua chunk: vim/_init_packages.lua:0: module 'vim.uri' not found:                                                                                                          │
│        no field package.preload['vim.uri']                                                                                                                                                          │
│        no file './vim/uri.lua'                                                                                                                                                                      │
│        no file '/home/runner/work/neovim/neovim/.deps/usr/share/luajit-2.1/vim/uri.lua'                                                                                                             │
│        no file '/usr/local/share/lua/5.1/vim/uri.lua'                                                                                                                                               │
│        no file '/usr/local/share/lua/5.1/vim/uri/init.lua'                                                                                                                                          │
│        no file '/home/runner/work/neovim/neovim/.deps/usr/share/lua/5.1/vim/uri.lua'                                                                                                                │
│        no file '/home/runner/work/neovim/neovim/.deps/usr/share/lua/5.1/vim/uri/init.lua'                                                                                                           │
│        no file './vim/uri.so'                                                                                                                                                                       │
│        no file '/usr/local/lib/lua/5.1/vim/uri.so'                                                                                                                                                  │
│        no file '/home/runner/work/neovim/neovim/.deps/usr/lib/lua/5.1/vim/uri.so'                                                                                                                   │
│        no file '/usr/local/lib/lua/5.1/loadall.so'                                                                                                                                                  │
│        no file './vim.so'                                                                                                                                                                           │
│        no file '/usr/local/lib/lua/5.1/vim.so'                                                                                                                                                      │
│        no file '/home/runner/work/neovim/neovim/.deps/usr/lib/lua/5.1/vim.so'                                                                                                                       │
│        no file '/usr/local/lib/lua/5.1/loadall.so'                                                                                                                                                  │
│stack traceback:                                                                                                                                                                                     │
│        [C]: in function 'require'                                                                                                                                                                   │
│        vim/_init_packages.lua: in function '__index'                                                                                                                                                │
│        vim/loader.lua: in function <vim/loader.lua:0>                                                                                                                                               │
│        [C]: at 0x5c7aa498b190                                                                                                                                                                       │
│        [C]: in function 'require'                                                                                                                                                                   │
│        vim/_init_packages.lua: in function '__index'                                                                                                                                                │
│        ...e/dhh/.local/share/nvim/lazy/lazy.nvim/lua/lazy/init.lua:61: in function 'setup'                                                                                                          │
│        /home/dhh/.config/nvim/lua/config/lazy.lua:10: in main chunk                                                                                                                                 │
│        [C]: in function 'require'                                                                                                                                                                   │
│        /home/dhh/.config/nvim/init.lua:2: in main chunk                                                                                                                                             │
│E484: Can't open file /usr/local/share/nvim/syntax/syntax.vim                                                                                                                                        │
```

**Comment by Salanoid** [comment]
I don't know exactly how to reproduce it, for me the installation of neovim v0.10.0 with lazygit worked with this script.

**Comment by dhh** [comment]
The installation worked fine, it's nvim itself at 0.10 that seems to have trouble with the lazyvim configuration.

**Comment by iljo-dp** [comment]
0.10 should work fine with previous configs unless you did some weird stuff, perhaps delete all the cache and data?

**Comment by Salanoid** [comment]
Yes, I use v0.10.0 with lazyvim and works fine.

**Comment by Salanoid** [comment]
I tried the script again on a fresh Ubuntu installation and encountered the same errors as @dhh. Neovim v0.10.0 from prebuilt doesn't work with Packer and LazyVim. To get the latest version with everything needed, we have to build it from source. I used [this old script of mine](https://github.com/Salanoid/install-bash-scripts/blob/master/install-nvim.sh), and it works fine.

**Comment by dhh** [comment]
Maybe we just need @tjdevries to release a 0.10.1, then? Either way, let's go the prebuilt binary way, but hold off until it's compatible with lazyvim etc.

**Comment by ealvar3z** [comment]
> > @Salanoid for me eza and alacritty (among others) are also missing from apt (no install candidate). Do you have any insight as to way that might be?
> 
> This thread might be helpful: [alacritty/alacritty#3863](https://github.com/alacritty/alacritty/issues/3863).
> 
> You can try running `apt-cache search alacritty` to look for the packages.
> 
> From what I've seen, you need to add an unofficial existing PPA `sudo add-apt-repository ppa:aslatter/ppa` to install Alacritty on different Ubuntu versions because it isn't available in the standard repositories. Or in this case, you can build Alacritty from the source https://github.com/alacritty/alacritty/blob/master/INSTALL.md.

You can always install the snap version with the `--classic` flag.

**Pull request merged**